x-pack/plugins/security_solution/public/common/components/sourcerer/index.tsx

```diff
@@ -251,49 +247,35 @@ export const Sourcerer = React.memo<SourcererComponentProps>(({ scope: scopeId }
           <EuiPopoverTitle data-test-subj="sourcerer-title">
             <>{i18n.SELECT_DATA_VIEW}</>
           </EuiPopoverTitle>
-          {isOnlyDetectionAlerts && (
-            <EuiCallOut
-              data-test-subj="sourcerer-callout"
-              iconType="iInCircle"
-              size="s"
-              title={isTimelineSourcerer ? i18n.CALL_OUT_TIMELINE_TITLE : i18n.CALL_OUT_TITLE}
-            />
-          )}
+          <SourcererCallout
+            isOnlyDetectionAlerts={isOnlyDetectionAlerts}
+            title={isTimelineSourcerer ? i18n.CALL_OUT_TIMELINE_TITLE : i18n.CALL_OUT_TITLE}
+          />
           <EuiSpacer size="s" />
-          {isModified === 'deprecated' || isModified === 'missingPatterns' ? (
-            <>
-              <TemporarySourcerer
-                activePatterns={activePatterns}
-                indicesExist={indicesExist}
-                isModified={isModified}
-                missingPatterns={missingPatterns}
-                onClick={resetDataSources}
-                onClose={setPopoverIsOpenCb}
-                onUpdate={isModified === 'deprecated' ? onUpdateDeprecated : onUpdateDataView}
-                selectedPatterns={selectedPatterns}
-              />
-              <UpdateDefaultDataViewModal
-                isShowing={isShowingUpdateModal}
-                missingPatterns={missingPatterns}
-                onClose={() => setIsShowingUpdateModal(false)}
-                onContinue={onContinueUpdateDeprecated}
-                onUpdate={onUpdateDataView}
-              />
-            </>
+          {(dataViewId === null && isModified === 'deprecated') ||
```

Comment:
**NOTE**: In this file, every change _except_ for the change to this condition to show `TemporarySourcerer` is simply moving code from `index.tsx` to `sub_components.tsx` to reduce arrow complexity